solana/restaking/programs/restaking/src/lib.rs

```diff
@@ -94,18 +94,37 @@ pub mod restaking {
 
         // Call Guest chain program to update the stake if the chain is initialized
         if guest_chain_program_id.is_some() {
+            let service =
+                service.as_ref().ok_or(error!(ErrorCodes::MissingService))?;
```

Comment:
Do we want lazy evaluation here?
```suggestion
                service.as_ref().ok_or_else(|| error!(ErrorCodes::MissingService))?;
```

Reply:
lazy evaluation sounds better to me.

Reply:
clippy seems to say the lazy evaluation is unneccesary so we can keep it as it is.